app/Http/Controllers/Auth/AuthController.php

```diff
@@ -31,22 +33,16 @@
 use Illuminate\Support\Facades\Mail;
 use Illuminate\Support\Facades\Validator;
 
-class UsersController extends AppController
+class AuthController extends AppController
 {
+    protected $password_recovery_resend_limit = 2 * 60; //2 minutes
+    protected $password_recovery_token_time = 120 * 60; // 2 hours
     protected $email_activation_token_time = 24 * 60 * 60; // 24 hours
     protected $email_activation_resend_limit = 15 * 60; // 15 minutes
     protected $email_change_token_time = 60 * 60; // 1 hour
     protected $email_new_change_resend_limit = 5 * 60; // 1 minute
 
-    /**
-     * Create a new controller instance.
-     */
-    public function __construct()
-    {
-        $this->middleware('snake_casing')->except(['emailChangeStep1']);
-    }
-
-    public function add(Request $request)
+    public function register(Request $request)
     {
         $this->validateRequestObject($request, 'user', User::$rules_add);
         Validator::make($request->all(), ['uri' => 'required'])->validate();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "required" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)